### PR TITLE
fix(panel): can't be reopened after being hidden

### PR DESCRIPTION
**Pull request by oliversalzburg**
## PR Checklist
Please check that your PR fulfills the following requirements:
- [X] The commit message follows [our guidelines](https://github.com/angular/material/blob/master/.github/CONTRIBUTING.md#-commit-message-format)
- [ ] Tests for the changes have been added or this is not a bug fix / enhancement
- [X] Docs have been added, updated, or were not required

## PR Type
What kind of change does this PR introduce?

```
[X] Bugfix
[ ] Enhancement
[X] Documentation content changes
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

When a panel is hidden once, it will not be displayed on screen when it is requested to be shown a second time. This is due to CSS classes still being applied to the panel after the animation finished.

Issue Number: #11291

## What is the new behavior?

The classes are correctly maintained and the panels can be reused many times.

## Does this PR introduce a breaking change?
```
[ ] Yes
[X] No
```

### CODE REVIEW

**Comment by oliversalzburg** [comment]
I also added a demo to showcase how to reuse panel instances and to easily observe the previously faulty behavior.

Can someone explain to me why everything falls apart if you use double-quotes in your demos instead of single-quotes? It seems to mess up the detection of the demo module name. This cost me quite some time and I didn't find the location in the docs build that dictates this style. Would appreciate if someone could point it out.

File: src/components/panel/demoReuse/index.html [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      <md-radio-button value="slide">Slide</md-radio-button>
+      <md-radio-button value="scale">Scale</md-radio-button>
+      <md-radio-button value="fade">Fade</md-radio-button>
+      <md-radio-button value="custom">Custom</md-radio-button>
+    </md-radio-group>
+  </div>
+
+  <div class="demo-md-panel-content">
+    <md-button class="md-primary md-raised demo-dialog-reuse-button"
+               ng-click="ctrl.showDialog($event,'Ahoi')">
```
Comment (Splaktar) [inline src/components/panel/demoReuse/index.html:18]:
```suggestion
               ng-click="ctrl.showDialog($event, 'Ahoi')">
```

File: src/components/panel/demoReuse/index.html [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      Open Dialog
+    </md-button>
+    <md-button class="md-accent md-raised demo-dialog-reuse-button"
+               ng-click="ctrl.showDialog($event,'Hello')">
```
Comment (Splaktar) [inline src/components/panel/demoReuse/index.html:22]:
```suggestion
               ng-click="ctrl.showDialog($event, 'Hello')">
```

File: src/components/panel/demoReuse/index.html [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      Also open Dialog
+    </md-button>
+    <md-button class="md-warning md-raised demo-dialog-reuse-button"
+               ng-click="ctrl.showDialog($event,'Greetings')">
```
Comment (Splaktar) [inline src/components/panel/demoReuse/index.html:26]:
```suggestion
               ng-click="ctrl.showDialog($event, 'Greetings')">
```

File: src/components/panel/demoReuse/style.global.css [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  transition: background 0.15s linear;
+  width: auto;
+}
+
+.demo-menu-item:hover,
+.demo-menu-item:focus {
+  background-color: rgb(238, 238, 238);
+}
+
+.demo-menu-item.selected {
```
Comment (Splaktar) [inline src/components/panel/demoReuse/style.global.css:53]: Most of the classes in this file don't seem to be used by this reuse demo?

File: src/components/panel/demoReuse/script.js [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+function ReusePanelCtrl(mdPanelRef, _demoCtrl) {
+  this._mdPanelRef = mdPanelRef;
+
+  // Register the controller for this panel with the parent controller.
+  _demoCtrl._mdPanelRefCtrl = this;
+}
+
+ReusePanelCtrl.prototype.closeDialog = function() {
+  var panelRef = this._mdPanelRef;
```
Comment (Splaktar) [inline src/components/panel/demoReuse/script.js:90]: It seems like we don't really benefit from this `var panelRef`?

**Pull request merged**